Feat/docs

eduardconstantin opened the pull request:
### Which issue(s) does this pull-request address?

Closes: #74 

### Description

Added index.html to docs folder to render documentation using docsify.
Added Readme.md as homepage.
Added _sidebar.md to have a custom sidebar.
This are the deployment steps from docsify docs: https://docsify.js.org/#/deploy
![image](https://github.com/user-attachments/assets/0cea8147-bfba-44e3-989b-901c2f14874d)
![image](https://github.com/user-attachments/assets/f617e458-bf70-4e41-b3c3-fe6849cc933a)
![image](https://github.com/user-attachments/assets/eef7a797-21ad-4c88-afd2-89076f04c503)


### Checklist

<!-- For completed items, change [ ] to [x]. -->

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [ ] Tests are included <!-- Bug fixes and new features should include tests -->
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

Tomas2D commented on docs/_sidebar.md:
```diff
@@ -0,0 +1,14 @@
+  - [Memory](memory.md)
+  - [Tools](tools.md)
+  - [Cache](cache.md)
+  - [SQL Tools](sql-tools.md)
```
I added a few more modules.

Tomas2D replied:
If possible, please put `SQL Tools` under `Tools`.

Tomas2D commented on docs/_sidebar.md:
```diff
@@ -0,0 +1,14 @@
+- Modules
+
+  - [Agents](agents.md)
+  - [LLMS](llms.md)
```
`LLMs`

Tomas2D reviewed (changes requested):
Great! Could you also please set up GitHub Action?

eduardconstantin commented:
@Tomas2D I added the action, I used this: https://github.com/marketplace/actions/docsify-action?version=v1.1
I also updated the sidebar
![image](https://github.com/user-attachments/assets/613b51f4-29f2-469a-9c46-338e6c624eb5)

Tomas2D reviewed (changes requested):
Great. However, there are some issues:

- It makes sense to leverage `https://github.com/docsifyjs/docsify-cli` so we could develop the docs in the watch mode (update `docs:build` script and add `docs:watch`)
- auto copy `README.md` to `docs/README.md` as part of the `docs:build`
- ensures that all code snippets are rendered correctly
- ensures that all links will point to the `https://i-am-bee.github.io/bee-agent-framework/` (if not developed locally)
- ensures that formatting [!NOTE] / [!TIP] and renders correctly because, from the provided screenshots, it does not.
- ensures that pipelines are passing (DCO / Lints)

Note: the `hacktoberfest` label is attached to the related issue; isn't that sufficient?

eduardconstantin commented:
ensures that formatting [!NOTE] / [!TIP] and renders correctly because, from the provided screenshots, it does not.
![image](https://github.com/user-attachments/assets/70fc5a3c-a380-40c4-9244-8a9e1c9697e0)

it seems that docsify needs pulgins to format different tags

eduardconstantin commented:
- [ ] It makes sense to leverage https://github.com/docsifyjs/docsify-cli so we could develop the docs in the watch mode (update docs:build script and add docs:watch) - **I added the docs:watch script, I don't know what to add to build**
- [ ] auto copy README.md to docs/README.md as part of the docs:build - **The links insed the readme file needs to be changed once the file is copied to the docs folder, but this step happened during `docsify init` command, if you call this command it will override the readme.md and index.html files (the index now has plugins added inside)**
- [x] ensures that all code snippets are rendered correctly
- [x] ensures that all links will point to the https://i-am-bee.github.io/bee-agent-framework/ (if not developed locally)
- [x] ensures that formatting [!NOTE] / [!TIP] and renders correctly because, from the provided screenshots, it does not.
- [x] ensures that pipelines are passing (DCO / Lints)

Tomas2D commented on package.json:
```diff
@@ -67,6 +67,7 @@
     "infra:clean-all": "yarn _docker compose down --volumes",
     "docs:build": "embedme --source-root=. docs/**/*.md",
     "docs:check": "yarn docs:build --verify",
+    "docs:watch": "docsify serve ./docs --open",
```
add `docsify` to dev dependencies

Tomas2D commented:
>I added the docs:watch script, I don't know what to add to build

`docsify generate` ?

eduardconstantin commented:
> > I added the docs:watch script, I don't know what to add to build
> 
> `docsify generate` ?

this command only generates the _sidebar.md file
![image](https://github.com/user-attachments/assets/819e628a-62db-4ea7-b4fe-a3dc543a3092)

Tomas2D reviewed (changes requested):
- A link under a code snippet is not working `Source: [examples/templates/objects.ts](http://localhost:3000/#/examples/templates/objects.ts)` (other links sometimes work and sometimes not, probably based on their path).
- Code is not highlighted
- When I click on a link within navbar, the target page shows, but the scroll position does not reset (so all of a sudden I am in the middle of a page instead of at its beginning)

eduardconstantin commented:
![image](https://github.com/user-attachments/assets/01db6475-3b35-42aa-aac2-8077fd2842cd)
Also fixed the rest of the issues

Tomas2D reviewed (changes requested):
You only changed paths, but those code snippets are generated dynamically and may vary from branch to branch (when versioning will be introduced).

Add this block to `window.$docsify`

```
    markdown: {
          smartypants: true,
          renderer: {
            link: function(link, renderer, text) {
              const example = link.match(/examples\/(.*)/)
              if (example) {
                link = "https://github.com/i-am-bee/bee-agent-framework/tree/main/examples/".concat(link)
              }
              return this.origin.link.apply(this, [link, renderer, text]);
            }
          }
        }
```

We will modify it in the future.

Tomas2D reviewed (changes requested):
Also, please rebase/update against the main and update README accordingly. 
Then, I will merge 🚀

Tomas2D reviewed (changes requested):
You forgot to remove `https://github.com/i-am-bee/bee-agent-framework/tree/main` prefixes`

eduardconstantin commented:
oh, yeah, sorry

Tomas2D reviewed (changes requested):
Not all links are updated.

Please update them and then squash all commits into one with the following text.

```
docs: init web documentation

Ref: i-am-bee/bee=agent-framework#74
```

eduardconstantin commented:
did it worked?
I used this command
`git merge --squash feature/login`

Tomas2D commented:
No problem, I will do it by myself. Thank you for your contribution.
